pkg/client/client_test.go

```diff
@@ -2199,1464 +2205,1765 @@ U5wwSivyi7vmegHKmblOzNVKA5qPO8zWzqBC
+				Expect(logs).To(ContainElement(MatchRegexp("Response Body.*DeploymentList")))
+			case !watchListEnabled && !watchListClientEnabled:
+				// Expect successful list
+				// List logs response body
+				Expect(logs).To(ContainElement(MatchRegexp("Response Body.*DeploymentList")))
+			}
+		})
 
-			It("should return an empty list if there are no matching objects", func() {
-				cl, err := client.New(cfg, client.Options{})
-				Expect(err).NotTo(HaveOccurred())
+		It("should fetch collection of objects, even if scheme is empty", func() {
+			By("create an initial object")
+			_, err := clientset.AppsV1().Deployments(ns).Create(ctx, dep, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			cl, err := client.New(cfg, client.Options{Scheme: runtime.NewScheme()})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("listing all objects of that type in the cluster")
+			deps := &unstructured.UnstructuredList{}
+			deps.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "apps",
+				Kind:    "DeploymentList",
+				Version: "v1",
+			})
+			err = cl.List(context.Background(), deps)
+			Expect(err).NotTo(HaveOccurred())
 
-				By("listing all Deployments in the cluster")
-				metaList := &metav1.PartialObjectMetadataList{}
-				metaList.SetGroupVersionKind(schema.GroupVersionKind{
+			Expect(deps.Items).NotTo(BeEmpty())
+			hasDep := false
+			for _, item := range deps.Items {
+				Expect(item.GroupVersionKind()).To(Equal(schema.GroupVersionKind{
 					Group:   "apps",
+					Kind:    "Deployment",
 					Version: "v1",
-					Kind:    "DeploymentList",
-				})
-				Expect(cl.List(context.Background(), metaList)).NotTo(HaveOccurred())
+				}))
+				if item.GetName() == dep.Name && item.GetNamespace() == dep.Namespace {
+					hasDep = true
+					break
+				}
+			}
+			Expect(hasDep).To(BeTrue())
+		})
 
-				By("validating no Deployments are returned")
-				Expect(metaList.Items).To(BeEmpty())
+		It("should return an empty list if there are no matching objects", func() {
+			cl, err := client.New(cfg, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("listing all Deployments in the cluster")
+			deps := &unstructured.UnstructuredList{}
+			deps.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "apps",
+				Kind:    "DeploymentList",
+				Version: "v1",
 			})
+			Expect(cl.List(context.Background(), deps)).NotTo(HaveOccurred())
 
-			// TODO(seans): get label selector test working
-			It("should filter results by label selector", func() {
-				By("creating a Deployment with the app=frontend label")
-				depFrontend := &appsv1.Deployment{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "deployment-frontend",
-						Namespace: ns,
-						Labels:    map[string]string{"app": "frontend"},
+			By("validating no Deployments are returned")
+			Expect(deps.Items).To(BeEmpty())
+		})
+
+		// TODO(seans): get label selector test working
+		It("should filter results by label selector", func() {
```

Comment:
Added this test as it was not implemented for unstructured before

Reply:
Can the TODO go then?

Reply:
Yup. Copied this from another case and forgot to look up why the TODO was there. Apparently when the TODO was added the test was commented out. Dropped the TODOs